src/components/Wizard/CreateVmWizard/CreateVmWizard.js

```diff
@@ -33,123 +26,70 @@ import {
+import { ConnectedVMWareImportProvider } from './providers/VMwareImportProvider/VMWareImportProvider';
 import { getVmWareProviderRequestedResources } from './providers/VMwareImportProvider/requestedResources';
-import { getVmSettings, getVmSettingValue, onCloseVmSettings } from './utils/vmSettingsTabUtils';
-
-const ALL_TAB_KEYS = [VM_SETTINGS_TAB_KEY, NETWORKS_TAB_KEY, STORAGE_TAB_KEY, RESULT_TAB_KEY];
+import { PROVIDER_VMWARE_V2V_NAME_KEY } from './providers/VMwareImportProvider/constants';
+import { get } from '../../../selectors';
+import { immutableListToShallowJS, concatImmutableLists } from '../../../utils/immutable';
 
-const getUpdatedValidatedState = (prevProps, prevState, props, state, extra) => {
-  const newState = ALL_TAB_KEYS.reduce(
-    (intermediaryState, tabKey) =>
-      getUpdatedState(tabKey, prevProps, prevState, props, intermediaryState, extra) || intermediaryState,
-    state
-  );
-
-  if (state === newState) {
-    return state; // skip validation
-  }
-
-  const validatedStepData = ALL_TAB_KEYS.reduce((resultStepData, tabKey) => {
-    const { value, valid } = newState.stepData[tabKey];
-    resultStepData[tabKey] = validateTabData(tabKey, value, valid, props);
-    return resultStepData;
-  }, {});
-
-  return {
-    ...newState,
-    stepData: validatedStepData,
-  };
-};
-
-export class CreateVmWizard extends React.Component {
+export class CreateVmWizardComponent extends React.Component {
   constructor(props) {
     super(props);
+    props.onInitialize();
 
-    const initialState = ALL_TAB_KEYS.reduce(
-      (state, tabKey) => {
-        state.stepData[tabKey] = getTabInitialState(tabKey, props);
-        return state;
-      },
-      {
-        activeStepIndex: getInitialActiveStepIndex(),
-        stepData: {},
-      }
-    );
-
-    this.state = getUpdatedValidatedState(null, null, props, initialState, {
-      // eslint-disable-next-line no-console
-      safeSetState: () => console.warn('setState not supported when initializing'),
-      virtualMachines: this.props.virtualMachines,
-    });
-  }
-
-  componentDidMount() {
-    this._unmounted = false;
+    this.state = {
+      activeStepIndex: 0,
+    };
   }
 
-  componentWillUnmount() {
-    this._unmounted = true;
-  }
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    let somePropChanged = false;
+    const changedProps = DETECT_PROP_CHANGES.reduce((changedPropsAcc, propName) => {
+      const propChanged = prevProps[propName] !== this.props[propName];
 
-  safeSetState = state => {
-    if (!this._unmounted) {
-      this.setState(state);
-    }
-  };
+      changedPropsAcc[propName] = propChanged;
 
-  componentDidUpdate(prevProps, prevState, snapshot) {
-    const newState = getUpdatedValidatedState(prevProps, prevState, this.props, this.state, {
-      safeSetState: this.safeSetState,
-      virtualMachines: this.props.virtualMachines,
-    });
+      if (!somePropChanged && propChanged) {
```

Comment:
Couldn't you just write this as `somePropChanged = !somePropChanged && propChanged;`? 

I prefer using the full if statement if the conditional expression is assigned a meaningful name to help the reader understand what is being tested more quickly.

Reply:
premature optimization, I guess it is not worth it (even though it is a didUpdate method)

changed for easier readability